src/logs/mod.rs

```diff
@@ -14,35 +17,54 @@ declare_log_buffer!(name = DEBUG_BUF, capacity = 1000);
 // Trace of HTTP requests and responses.
 declare_log_buffer!(name = TRACE_HTTP_BUF, capacity = 1000);
 
-pub const INFO: PrintProxySink = PrintProxySink("INFO", &INFO_BUF);
-pub const DEBUG: PrintProxySink = PrintProxySink("DEBUG", &DEBUG_BUF);
-pub const TRACE_HTTP: PrintProxySink = PrintProxySink("TRACE_HTTP", &TRACE_HTTP_BUF);
+pub const INFO: PrintProxySink = PrintProxySink(&LogMessageType::Info, &INFO_BUF);
+pub const DEBUG: PrintProxySink = PrintProxySink(&LogMessageType::Debug, &DEBUG_BUF);
+pub const TRACE_HTTP: PrintProxySink = PrintProxySink(&LogMessageType::TraceHttp, &TRACE_HTTP_BUF);
 
 #[derive(Debug)]
-pub struct PrintProxySink(&'static str, &'static GlobalBuffer);
+pub struct PrintProxySink(&'static LogMessageType, &'static GlobalBuffer);
 
 impl Sink for PrintProxySink {
     fn append(&self, entry: ic_canister_log::LogEntry) {
-        ic_cdk::println!("{} {}:{} {}", self.0, entry.file, entry.line, entry.message);
+        let message_type = self.0;
+        if get_log_message_filter().should_print_log_message(*message_type) {
```

Comment:
We should also add unit tests that the expected messages are logged/filtered out.

Reply:
Done. Let me know if you think of any other tests that would be good to include.